@xen-orchestra/lite/src/components/Console.tsx

```diff
@@ -68,9 +80,45 @@ const Console = withState<State, Props, Effects, Computed, ParentState, ParentEf
       _handleConnect: function () {
         this.state.rfbConnected = true
       },
-      _handleDisconnect: function () {
+      _handleDisconnect: async function () {
         this.state.rfbConnected = false
-        this.effects._connect()
+        const { protocol } = window.location
+        if (protocol === Protocols.https) {
+          try {
+            await fetch(`${protocol}//${this.state.url?.host}`)
+          } catch (error) {
+            console.error(error)
+            try {
+              await confirm({
+                icon: 'exclamation-triangle',
+                message: (
+                  <div>
+                    <p>
+                      <IntlMessage
+                        id='unreachableHost'
+                        values={{
+                          name: this.state.objectsByType
+                            .get('host')
+                            ?.find(host => host.address === this.state.url?.host)?.name_label,
+                        }}
+                      />
+                    </p>
+                    <a href={`${protocol}//${this.state.url?.host}`} rel='noopener noreferrer' target='_blank'>
+                      {this.state.url?.host}
+                    </a>
+                  </div>
+                ),
+                title: <IntlMessage id='connectionError' />,
+              })
+            } catch {
+              this.state.tryToReconnect = false
+            }
+          }
+        }
+
+        if (this.state.tryToReconnect) {
+          this.effects._connect()
```

Comment:
You can put this in the second `try/catch`

Reply:
No. In case we are not using `HTTPS` protocol we want to reconnect. ( current behaviour ).
The `try/catch` is only executed in `HTTPS` case